packages/core-dialog/core-dialog.js

```diff
@@ -1 +1,128 @@
+  const options = typeof open === 'object' ? open : {open}
+
+  return queryAll(dialogs).forEach((dialog) => {
+    dialog.setAttribute(UUID, '')
+    dialog.setAttribute('role', 'dialog')
+    dialog.setAttribute('aria-modal', true)
+    dialog.setAttribute('aria-label', dialog.getAttribute('aria-label') || options.label)
+
+    toggleDialog(dialog, options.open)
+    return dialog
+  })
+}
+
+addEvent(UUID, 'click', (event) => {
+  for (let el = event.target, isClose; el; el = el.parentElement) {
+    const action = el.getAttribute('data-core-dialog')
+    isClose = isClose || action === 'close'
+    if (isClose) el.hasAttribute(UUID) && dialog(el, false)
+    else if (action) dialog(action, true)
+  }
+})
+
+addEvent(UUID, 'keydown', (event) => {
+  if (event.keyCode === KEYS.TAB) keepFocus(event)
+  if (event.keyCode === KEYS.ESC) {
+    if (!event.defaultPrevented) dialog(getTopLevelDialog(), false)// event.preventDefault()
+    event.preventDefault()
+    /* else {
+      event.preventDefault()
+      dialog(getTopLevelDialog(), false)
+    } */
+  }
+})
+
+const getZIndexOfElement = (element) =>
+  Number(window.getComputedStyle(element).getPropertyValue('z-index')) || 0
+
+// Find the last focused element before opening the dialog
+const getLastFocusedElement = () =>
+  queryAll(`[${PREVIOUS}]`).sort((a, b) =>
+    a.getAttribute(PREVIOUS) < b.getAttribute(PREVIOUS)
+  )[0] || document.body
+
+const getTopLevelDialog = () =>
+  queryAll(`[${UUID}][open]`).sort((a, b) =>
+    getZIndexOfElement(a) < getZIndexOfElement(b)
+  )[0]
+
+function toggleDialog (dialog, open) {
+  const last = getLastFocusedElement()
+  const isOpen = toggleOpen(dialog, open)
+  const lastIndex = Number(last.getAttribute(`${PREVIOUS}`)) || 0
+  const topZIndex = Math.max(...queryAll('*').map(getZIndexOfElement))
+
+  if (isOpen) {
+    const focusable = queryFocusable(dialog)[0]
+    document.activeElement.setAttribute(`${PREVIOUS}`, lastIndex + 1)
+    dialog.style.zIndex = topZIndex + 2
+    focusable && focusable.focus()
+
+    dialog.insertAdjacentElement('afterend', BACKDROP)
+    BACKDROP.removeAttribute('hidden')
+    BACKDROP.style.zIndex = topZIndex + 1
+  } else {
+    const topDialog = getTopLevelDialog()
+    if (topDialog) BACKDROP.style.zIndex = topDialog.style.zIndex - 1
+    else BACKDROP.setAttribute('hidden', '')
+    last.removeAttribute(`${PREVIOUS}`)
+    last.focus()
+  }
+}
+
+function toggleOpen (dialog, open) {
+  const prevState = dialog.hasAttribute('open')
+  const nextState = typeof open === 'boolean' ? open : (open === 'toggle' ? !prevState : prevState)
+  const canUpdate = prevState === nextState || dispatchEvent(dialog, 'dialog.toggle', {isOpen: prevState, willOpen: nextState})
+
+  if (canUpdate) {
+    if (SUPPORT) {
+      dialog.open = !nextState
+      dialog[nextState ? 'show' : 'close']()
+    } else dialog[nextState ? 'setAttribute' : 'removeAttribute']('open', '') // Toggle open attribute
+  }
```

Comment:
OCD 🎉  this can become:
```
if (canUpdate) {
  if (SUPPORT) dialog[(dialog.open = !nextState) ? 'close' : 'show']()
  else dialog[nextState ? 'setAttribute' : 'removeAttribute']('open', '')
}
```

Reply:
It might look nicer, but it's harder to read. Don't think we should have an assignment in a ternary operation either.